tests/test_load.py

```diff
@@ -67,3 +70,96 @@ def test_store_table_error(syn):
     with patch.object(syn, "store", side_effect=SynapseTimeoutError) as patch_store:
         load.store_table(syn, "full/path", "syn1234")
         patch_store.assert_called_once()
+
+
+def test__update_table_non_empty_dataframe(syn):
+    """Test _update_table function with both new_dataset and database as non_empty dataframe"""
+    database_synid = "syn123"
+    primary_key_cols = ["test", "foo"]
+    to_delete = (False,)
```

Comment:
Because the `_update_table` function has `to_delete` as a bool, it would be good to have a test case where `to_delete` is True and you can mock `process_functions._delete_rows` and call `assert_called_once_with` on it so you can make sure it's called with the expected arguments

Reply:
Will check on this. Thanks for catching this.